pandas/core/algorithms.py

```diff
@@ -1301,12 +1306,13 @@ def compute(self, method: str) -> Series:
 
         if self.keep != "all":
             inds = inds[:n]
+            findex = nbase
 
         if self.keep == "last":
             # reverse indices
             inds = narr - 1 - inds
 
-        return dropped.iloc[inds]
+        return concat([dropped.iloc[inds], nan_index])[:findex]
```

Comment:
dont' you need to use .iloc here? as this will not be a positional lookup (it might work with a range index though)

Reply:
Should be done now